lib/riak/search/result_collection.rb

```diff
@@ -0,0 +1,53 @@
+require 'riak/search'
+require 'ostruct'
+
+module Riak::Search
+  class ResultCollection
+    attr_reader :client
+    attr_reader :raw
+    attr_reader :max_score
+    attr_reader :length
+    attr_reader :num_found
+
+    def initialize(client, raw_results)
+      @client = client
+      @raw = raw_results
+      @max_score = raw['max_score']
+      @num_found = raw['num_found']
+      @length = raw['docs'].length
+    end
+
+    def docs
+      @docs ||= raw['docs'].map do |result|
+        ResultDocument.new client, result
+      end
+    end
+
+    def empty?
+      length == 0
+    end
+
+    def [](index)
+      doc = docs[index]
+      return nil if doc.nil?
+
+      doc.robject ||= doc.bucket.get doc.key
```

Comment:
This seems duplicated in `ResultDocument`. Why is it necessary to prefetch here?